plugins/logs/plugin_test.go

```diff
@@ -791,7 +840,7 @@ type testFixture struct {
 	server  *testServer
 }
 
-func newTestFixture(t *testing.T) testFixture {
+func newTestFixture(t *testing.T, pluginConfigRaw string) testFixture {
```

Comment:
nit: we might want to change this option to be more of a like overlay or additional config. Making the callers provide `{"service": "example"}` is kind of of hassle, especially when that service is being defined automatically in this helper. Any change to it may require the callers to update.

A pattern we use a bunch in OPA is to allow for optionally passing in config functions like:

```go
type testPluginCustomizer func(c *Config)
```
and then to customize the config for a test have something like:
```go
func newTestFixture(t *testing.T, options ...testPluginCustomizer) testFixture {
	...
	config, _ := ParseConfig([]byte(pluginConfig), manager.Services(), nil)
	for _, option := range options {
		option(config)
	}
	...
}
```
called with something like:
```go
	fixture := newTestFixture(t, func(c *Config) {
		c.Reporting.UploadSizeLimitBytes = 300
	})
```

All the other callers can skip providing anything so they stay as `newTestFixture(t)`

The only potential downside of that approach is that it skips the `validateAndInjectDefaults` step, but I'm guessing thats OK for unit tests where the test author is setting very explicit options.

Reply:
Oh, nice! I was reaching for something like optional/default parameters for this exactly but came up blank - I've never done much golang except for some small fixes here and there, so learning patterns like this is awesome. I'll update the PR with that once I get a few moments to spare sometime this week. Thanks @patrick-east 👍